### PR TITLE
Fix several ResourceWarnings in test_requests.py

### PR DESCRIPTION
**Pull request by BoboTiG**
Hello,

I do not know if you are interested in fixing `ResourceWarning`s in tests, but I will take my chance :)
There was 10 unclosed files in `test_requests.py`.

### CODE REVIEW

File: tests/test_requests.py [hunk 1]
```diff
@@ -875,11 +875,12 @@ def test_form_encoded_post_query_multivalued_element(self, httpbin):
...
-            data={'stuff': json.dumps({'a': 123})},
-            params={'blah': 'asdf1234'},
-            files={'file': ('test_requests.py', open(__file__, 'rb'))})
-        assert r.status_code == 200
+        with open(__file__, 'rb') as f:
+            r = requests.post(httpbin('post'),
+                data={'stuff': json.dumps({'a': 123})},
+                params={'blah': 'asdf1234'},
+                files={'file': ('test_requests.py', f)})
+            assert r.status_code == 200
```
Comment (nateprewitt) [inline tests/test_requests.py]: Let’s pull the `assert` out to the same indentation level as `with`.

File: tests/test_requests.py [hunk 2]
```diff
@@ -889,37 +890,41 @@ def test_different_encodings_dont_break_post(self, httpbin):
...
     def test_unicode_multipart_post(self, httpbin, data):
-        r = requests.post(httpbin('post'),
-            data=data,
-            files={'file': ('test_requests.py', open(__file__, 'rb'))})
-        assert r.status_code == 200
+        with open(__file__, 'rb') as f:
+            r = requests.post(httpbin('post'),
+                data=data,
+                files={'file': ('test_requests.py', f)})
+            assert r.status_code == 200
```
Comment (nateprewitt) [inline tests/test_requests.py]: Same here.

File: tests/test_requests.py [hunk 3]
```diff
@@ -889,37 +890,41 @@ def test_different_encodings_dont_break_post(self, httpbin):
...
+            assert b'name="stuff"' in prep.body
+            assert b'name="b\'stuff\'"' not in prep.body
 
     def test_unicode_method_name(self, httpbin):
-        files = {'file': open(__file__, 'rb')}
-        r = requests.request(
-            method=u('POST'), url=httpbin('post'), files=files)
-        assert r.status_code == 200
+        with open(__file__, 'rb') as f:
+            files = {'file': f}
```
Comment (nateprewitt) [inline tests/test_requests.py]: Let's just pass this directly to the `files` parameter.

File: tests/test_requests.py [hunk 4]
```diff
@@ -889,37 +890,41 @@ def test_different_encodings_dont_break_post(self, httpbin):
...
+            r = requests.request(
+                method=u('POST'), url=httpbin('post'), files=files)
+            assert r.status_code == 200
```
Comment (nateprewitt) [inline tests/test_requests.py]: We can unindent this to align with `with`.

File: tests/test_requests.py [hunk 5]
```diff
@@ -889,37 +890,41 @@ def test_different_encodings_dont_break_post(self, httpbin):
...
-        prep = r.prepare()
-        assert b'name="stuff"' in prep.body
-        assert b'name="b\'stuff\'"' not in prep.body
+        with open(filename, 'rb') as f:
+            r = requests.Request(
+                method='POST', url=httpbin('post'),
+                data={'stuff'.encode('utf-8'): 'elixr'},
+                files={'file': ('test_requests.py', f)})
+            prep = r.prepare()
+            assert b'name="stuff"' in prep.body
```
Comment (nateprewitt) [inline tests/test_requests.py]: These asserts can be done at the same indentation as the `with` block.

**Comment by BoboTiG** [comment]
@nateprewitt, thanks for the review. You are totally right and I applied changes + rebase on master :)

We only need to keep `r.prepare*()` at the same indentation level as the definition of `r` to prevent `models.py:159: ValueError: read of closed file`.

**Comment by BoboTiG** [comment]
What is the status of the PR? Would you mind trigger the request-CI job?

**Comment by BoboTiG** [comment]
Ping :)

**Review by nateprewitt — APPROVED** [review]
Hi @BoboTiG, I've rebased your change onto the top of our main branch and resolved some conflicts. This seems like a reasonable fix still and I think we're happy with its current state. Thanks again for the contribution!

**Pull request merged**